File: PKIService/bezbednost-backend/src/main/java/com/example/bezbednostbackend/dto/EmployeeDTO.java
package com.example.bezbednostbackend.dto;

import com.example.bezbednostbackend.model.Role;
import com.example.bezbednostbackend.model.User;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class EmployeeDTO {

    private Integer id;
    @NotEmpty(message = "Name is required")
    @Size(max = 100, message = "Name cannot have more than a 100 characters")
    private String name;
    @NotEmpty(message = "Surname is required")
    @Size(max = 100, message = "Surname cannot have more than a 100 characters")
    private String surname;
    @NotEmpty(message = "Username is required")
    @Size(max = 100, message = "Username cannot have more than a 100 characters")
    private String username;
    @NotEmpty(message = "Password is required")
    @Size(max = 100, message = "Password cannot have more than a 100 characters")
    private String password;
    @NotEmpty(message = "Address id is required")
    private Integer addressID;
    @NotEmpty(message ="Phone number is required")
    @Size(max = 100, message = "Phone number cannot have more than a 100 characters")
    private String phoneNumber;
    @NotEmpty(message = "Role is required")
    @Size(max = 100, message = "Role cannot have more than a 100 characters")
    private String role;
    @NotEmpty(message = "Work title is required")
    @Size(max = 100, message = "Work title cannot have more than a 100 characters")
    private String workTitle;
    private String dateOfEmployment;

    private boolean isBlocked;
    private boolean allowRefreshToken;

    public EmployeeDTO(User dto){

        for(Role role : dto.getRoles()){
            this.role = role.getName();
        }
        this.id = dto.getId();
        this.name = dto.getName();
        this.surname = dto.getSurname();
        this.username = dto.getUsername();
        this.password = dto.getPassword();
        this.addressID = dto.getAddress().getId();
        this.phoneNumber = dto.getPhoneNumber();
        this.workTitle = dto.getWorkTitle();
        this.isBlocked=dto.isBlocked();
        this.allowRefreshToken=dto.isAllowRefreshToken();
        this.dateOfEmployment = dto.getStartOfEmployment().toString();
    }

}

File: PKIService/pki-service-app/src/app/model/employee-dto.ts
export class EmployeeDTO {
    id: number = 0
    name: string = ''
    surname: string = ''
    username: string = ''
    password: string = ''
    addressID: number = 0
    phoneNumber: string = ''
    role: string = ''
    isBlocked:boolean=false;
    allowRefreshToken:boolean=true;
    dateOfEmployment = ''

    public constructor(obj?: any) {
        if (obj) {
            this.id = obj.id;
            this.name = obj.name;
            this.surname = obj.surname;
            this.username = obj.username;
            this.password = obj.password;
            this.addressID = obj.addressID;
            this.phoneNumber = obj.phoneNumber;
            this.role = obj.role;
            this.isBlocked=obj.isBlocked;
            this.allowRefreshToken=obj.allowRefreshToken;
            this.dateOfEmployment = obj.dateOfEmployment
        }
    }

}
